### PR TITLE
[Backport 1.20] - fix possible race where bootstrap data might not save

### PR DESCRIPTION
**Pull request by briandowns**
Signed-off-by: Brian Downs <brian.downs@gmail.com>

#### Types of Changes ####

Backport from original issue #3414 

This required backporting supporting code for this fix.

#### Linked Issues ####

#3484 
#3546

### CODE REVIEW

**Review by brandond — APPROVED** [review]

File: Dockerfile.dapper096384309 [hunk 1]
```diff
@@ -0,0 +1,2 @@
+FROM k3s:issue-3234
```
Comment (galal-hussein) [inline Dockerfile.dapper096384309:1]: I think that file was wrongly commited

File: Dockerfile.dapper203010478 [hunk 2]
```diff
@@ -0,0 +1,53 @@
+ARG GOLANG=golang:1.16.2-alpine3.12
```
Comment (galal-hussein) [inline Dockerfile.dapper203010478:1]: Same here, I dont think this file should be checked in

File: Dockerfile.dapper096384309 (hunk 1 above)
Comment (briandowns) [reply Dockerfile.dapper096384309]: Fixed

File: Dockerfile.dapper203010478 (hunk 2 above)
Comment (briandowns) [reply Dockerfile.dapper203010478]: Fixed

**Review by brandond — CHANGES_REQUESTED** [review]
I feel like this should be linked to an issue for backporting the snapshot delete functionality, along with any other features or bugfixes that are getting pulled in, otherwise QA won't know what all to test.

**Comment by briandowns** [comment]
> I feel like this should be linked to an issue for backporting the snapshot delete functionality, along with any other features or bugfixes that are getting pulled in, otherwise QA won't know what all to test.

Issue added and linked.

**Review by galal-hussein — APPROVED** [review]

**Review by brandond — APPROVED** [review]

**Pull request merged**